craft_parts/executor/part_handler.py

```diff
@@ -1146,6 +1145,39 @@ def _clean_shared(
             )
             overlay_migration_state_path.unlink()
 
+    def _symlink_alias_to_default(self) -> None:
+        """Create directory and symlinks for the alias of the default partition.
+
+        These symlinks are never consumed by craft-parts. They are created to help
+        users debugging a build.
+        """
+        if not self._part_info.is_default_partition_aliased:
+            return
+        default_partition = self._part_info.default_partition
+        logger.debug("Create symlinks for %s", default_partition)
+        self._part_info.alias_partition_dir.mkdir(parents=True, exist_ok=True)
+
+        for src, dst in [
+            (self._part_info.parts_dir, self._part_info.parts_alias_symlink),
+            (self._part_info.stage_dir, self._part_info.stage_alias_symlink),
+            (self._part_info.prime_dir, self._part_info.prime_alias_symlink),
+            (self._part_info.overlay_dir, self._part_info.overlay_alias_symlink),
+        ]:
+            if dst.exists():
+                if not dst.is_symlink():
+                    # Between two runs of the lifecycle, the default partition alias name
+                    # can be changed to a previously concrete partition by the user.
+                    raise RuntimeError(
+                        f"cannot create symlinks {dst}, a concrete directory already exists."
```

Comment:
@mr-cal I am not sure about this error handling. I am tempted to advise the user to clean everything but  it sounds wrong. What do you think?

Reply:
Looking at the other runtime errors in craft_parts/executor/, I agree with you.

I'm not sure if we should suggest to clean. If this error happens unexpectedly and cleaning works, the user may not report the bug.

Reply:
I defer to Alex on this one, unless we're uncertain about the wording.